### PR TITLE
Fix required form radio/checkbox fields

### LINKED ISSUE #1517: Required checkbox not working
Describe the bug:
I should be able to add the checkbox to the contact form so that form can't be sent out without checking it. However, the form is sent out even if this required field is not checked.

To reproduce:

1. Create a Form using the first variation.
2. Add the checkbox field before the Submit button.
3. Make the checkbox field required.
4. Submit the form.

The form should alert that the required field is empty.
Instead, the form is being sent out.

- [X] This bug happens with no other plugins activated
- [X] This bug happens with a default WordPress theme active
- [X] This bug happens **without** the Gutenberg plugin active
- [ ] I can reproduce this bug consistently using the steps above

### WordPress version: 5.4.1

### LINKED ISSUE #1653: ISNBAT render a Form which returns errors
### Describe the bug:

Using the following markup we can generate a form within the editor. When puslished and filled from the front-end we are seeing error messages. 

```
<!-- wp:coblocks/form -->
<!-- wp:coblocks/field-name /-->

<!-- wp:coblocks/field-email {"required":true} /-->

<!-- wp:coblocks/field-radio {"options":["Option"]} /-->

<!-- wp:coblocks/field-select {"options":["Option"]} /-->

<!-- wp:coblocks/field-radio {"options":["Option"]} /-->

<!-- wp:coblocks/field-textarea {"required":true} /-->

<!-- wp:coblocks/field-submit-button {"submitButtonText":"Contact Us"} /-->
<!-- /wp:coblocks/form -->
```

**Errors**
```html
Notice: Undefined index: value in /Users/aledesma/development/wordpress/wp-content/plugins/coblocks/includes/class-coblocks-form.php on line 887

Notice: Undefined index: value in /Users/aledesma/development/wordpress/wp-content/plugins/coblocks/includes/class-coblocks-form.php on line 893

Notice: Undefined index: label in /Users/aledesma/development/wordpress/wp-content/plugins/coblocks/includes/class-coblocks-form.php on line 893

Notice: Undefined index: value in /Users/aledesma/development/wordpress/wp-content/plugins/coblocks/includes/class-coblocks-form.php on line 887

Notice: Undefined index: value in /Users/aledesma/development/wordpress/wp-content/plugins/coblocks/includes/class-coblocks-form.php on line 893

Notice: Undefined index: label in /Users/aledesma/development/wordpress/wp-content/plugins/coblocks/includes/class-coblocks-form.php on line 893
```

### To reproduce:

1. Generate the block by copy/pasting the markup provided above.
2. Publish post and view page from front-end.
3. Complete form and submit.
4. See error.

### Expected behavior:

Expect improperly setup blocks not to render on front-end.
Expect improperly blocks setup within form show an error to end-user specifying what is wrong.

### Isolating the problem:

- [x] This bug happens with no other plugins activated
- [x] This bug happens with a default WordPress theme active
- [x] This bug happens **without** the Gutenberg plugin active
- [x] I can reproduce this bug consistently using the steps above

### WordPress version:

5.5
### Gutenberg version:

None

### PR DESCRIPTION
**Pull request by EvanHerman**
### Note
If using Go to test, this will require https://github.com/godaddy-wordpress/go/pull/579

### Description
Made sure that the checkboxes/radio fields of form block, when set to required, prevent the form from being submitted until at least one option/checkbox is selected.

Resolves #1517 and resolves #1653.

**Note:** HTML does not provide a way to require checkbox groups. The workaround was to enqueue a script and display a notice above the field when no checkboxes were checked.

**The notice:** `Please select at least one checkbox.`
**Wrapped in the filter:** `coblocks_form_checkbox_required_text`

### Screenshots
form-checkboxes

### Types of changes
Bug fix (non-breaking change which fixes an issue)

### How has this been tested?
Manually

### Checklist:
- [x] My code is tested
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: includes/class-coblocks-form.php [hunk 1]
```diff
@@ -588,10 +590,37 @@ public function render_field_checkbox( $atts ) {
...
+			$required ? esc_attr( ' required' ) : '',
+			$required ? sprintf(
+				'<div class="required-error hidden">%s</div>',
+				/**
+				 * Filter the checkbox required text that displays when no checkbox is
+				 * selected when the form is submitted.
+				 *
+				 * @param string $error_text Error text displayed to the user.
+				 */
+				(string) apply_filters( 'coblocks_form_checkbox_required_text', esc_html__( 'Please select at least one checkbox.', 'coblocks' ) )
```
Comment (AnthonyLedesma) [inline includes/class-coblocks-form.php:621]: If we are adding a new filter then we should also update the Docs to include information on this filter and how to apply it.

File: src/js/coblocks-checkbox-required.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		submittedForm.find( '.required-error' ).hide();
+	} );
+
+	/**
+	 * Toggle the `.required-error` visibility when a checkbox is selected
+	 */
+	$( 'body' ).on( 'change', '.coblocks-field.checkbox.required input[type="checkbox"]', function() {
+		$( this ).closest( '.coblocks-field.checkbox.required' ).find( '.required-error' ).hide();
+	} );
+} );
```
Comment (AnthonyLedesma) [inline src/js/coblocks-checkbox-required.js:30]: Ideally, we should avoid using jQuery in a case like this. `show`, `hide`, and `find` can all be replicated with vanilla javascript without needing to load jQuery.

File: includes/class-coblocks-form.php [hunk 3]
```diff
@@ -588,10 +590,37 @@ public function render_field_checkbox( $atts ) {
...
 		$label      = isset( $atts['label'] ) ? $atts['label'] : __( 'Select', 'coblocks' );
 		$label_slug = $checkbox_count > 1 ? sanitize_title( $label . '-' . $checkbox_count ) : sanitize_title( $label );
+		$required   = ( isset( $atts['required'] ) && $atts['required'] );
+
+		if ( $checkbox_count === 1 && $required ) {
+
+			wp_enqueue_script(
+				'coblocks-checkbox-required',
+				CoBlocks()->asset_source( 'js' ) . 'coblocks-checkbox-required.js',
+				array( 'jquery' ),
```
Comment (AnthonyLedesma) [inline includes/class-coblocks-form.php:600]: I left another comment regarding keeping jQuery as a dependency here. In the interest of continuing to work toward https://github.com/godaddy-wordpress/coblocks/issues/1365, we should ideally avoid the use of jQuery all together.

File: .dev/tests/phpunit/includes/test-coblocks-form.php [hunk 4]
```diff
@@ -508,6 +508,29 @@ public function test_render_field_checkbox_inline() {
...
+					'option-2' => 'Option 2',
+				],
+				'required' => true
+			],
+			''
+		);
+
+		global $wp_scripts;
+
+		$this->assertArrayHasKey( 'coblocks-datepicker', $wp_scripts->registered );
```
Comment (AnthonyLedesma) [inline .dev/tests/phpunit/includes/test-coblocks-form.php:530]: Should be `'coblocks-checkbox-required'` not `'coblocks-datepicker'`

**Comment by EvanHerman** [comment]
@AnthonyLedesma Good for another review 👌

**Comment by AnthonyLedesma** [comment]
Confirmed this closes #1653.

**Review by AnthonyLedesma — CHANGES_REQUESTED** [review]
We are very close. The required attributes seem to be properly preventing form submission but we have a case where no errors are shown to the user. To replicate this bug use the following markup. 
```
<!-- wp:coblocks/form -->
<!-- wp:coblocks/field-email {"required":true} /-->

<!-- wp:coblocks/field-radio {"required":true,"options":["Yes","No"]} /-->

<!-- wp:coblocks/field-checkbox {"required":true,"options":["Yes","No"]} /-->

<!-- wp:coblocks/field-textarea {"required":true} /-->

<!-- wp:coblocks/field-submit-button {"submitButtonText":"Contact Us"} /-->
<!-- /wp:coblocks/form -->
```

When you leave the radio options unchecked and fill all other form elements, we have a case where form fails to submit and no errors are shown to the user. There should be an error message here specifying that an option must be selected.

**Comment by EvanHerman** [comment]
@AnthonyLedesma This was a bug with Go, setting the radio buttons to `appearance: none;`

Feel free to pull down https://github.com/godaddy-wordpress/go/pull/579 and test, or use a different theme.

**Review by AnthonyLedesma — APPROVED** [review]
Confirmed that the bug reported was with Go.

**Pull request merged**